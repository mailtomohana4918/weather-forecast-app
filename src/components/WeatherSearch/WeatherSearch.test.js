import { render,fireEvent } from "@testing-library/react";
import React from "react";
import {WeatherSearch} from "./index";
import { act } from "react-dom/test-utils";
import axios from 'axios'


jest.mock('axios')

const weatherApiResponse = { 
    data: {
        location: {
            name: "Warrington",
            region: "Buckinghamshire",
            country: "United Kingdom",
            lat: 52.18,
            lon: -0.69,
            tz_id: "Europe/London",
            localtime_epoch: 1660168714,
            localtime: "2022-08-10 23:01"
        },
        current: {
            last_updated_epoch: 1660167900,
            last_updated: "2022-08-10 22:45",
            temp_c: 22,
            temp_f: 71.6,
            is_day: 0,
            condition: {
                text: "Clear",
                icon: "//cdn.weatherapi.com/weather/64x64/night/113.png",
                code: 1000
            },
            wind_mph: 5.6,
            wind_kph: 9,
            wind_degree: 100,
            wind_dir: "E",
            pressure_mb: 1024,
            pressure_in: 30.24,
            precip_mm: 0,
            precip_in: 0,
            humidity: 53,
            cloud: 0,
            feelslike_c: 24.5,
            feelslike_f: 76,
            vis_km: 10,
            vis_miles: 6,
            uv: 1,
            gust_mph: 13.6,
            gust_kph: 22
        }
    },
    status: 200,
}

describe("WeatherSearch Component", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<WeatherSearch />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("form validation error when user not entered any one of the input fields", () => {
    axios.get.mockResolvedValueOnce(weatherApiResponse)
    const { getByTestId } = render(<WeatherSearch />);
    const searchButton = getByTestId("search-button-test-id");
    fireEvent.click(searchButton)
    const cityTextInput = getByTestId("city-test-id");
    const countryTextInput = getByTestId("country-test-id");
    expect(cityTextInput.getElementsByClassName("MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl")).toBeTruthy()
    expect(countryTextInput.getElementsByClassName("MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl")).toBeTruthy()
  });
  it("form success when user entered the input and click on search",() => {
    const { getByTestId} = render(<WeatherSearch />);
    const cityTextInput = getByTestId("city-test-id");
    fireEvent.change(cityTextInput, { target: { value: "London" } });
    const searchButton = getByTestId("search-button-test-id");
    act(()=>{
        fireEvent.click(searchButton)
    })
    expect(cityTextInput.getElementsByClassName("MuiInputBase-input MuiOutlinedInput-input")).toBeTruthy()
    
  });
});

