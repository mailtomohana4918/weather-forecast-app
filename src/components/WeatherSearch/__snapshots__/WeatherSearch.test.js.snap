// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherSearch Component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
          >
            Weather Portal
          </h5>
        </div>
      </div>
    </header>
    <form>
      <div
        class="MuiBox-root MuiBox-root-4 makeStyles-boxContainer-2"
      >
        <div
          class="MuiBox-root MuiBox-root-5 makeStyles-textBoxContainer-1"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="city-test-id"
                  name="city"
                  placeholder="Enter city"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-8"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-11 makeStyles-textBoxContainer-1"
        >
          <div
            class="MuiBox-root MuiBox-root-12"
          >
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="country-test-id"
                  maxlength="60"
                  name="country"
                  placeholder="Enter country"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-8"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-13 makeStyles-textBoxContainer-1"
        >
          <div
            class="MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-15 Mui-checked MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-17"
                    name="mui-31066"
                    type="radio"
                    value="celsius"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-18 PrivateRadioButtonIcon-checked-20"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-19"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Celsius
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-17"
                    name="mui-31066"
                    type="radio"
                    value="fahrenheit"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-18"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-19"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Fahrenheit
              </span>
            </label>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-21 makeStyles-butonContainer-3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="search-button-test-id"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
  >
     
  </div>
</DocumentFragment>
`;
